Tickets/dm 1763

danielsf opened the pull request.

danielsf committed:
added all of the necessary code

danielsf committed:
removed unneeded print statements

danielsf committed:
removed unneeded quotation marks

danielsf committed:
cleaned up patch file

ktlim commented on patches/tmvTable.patch:
```diff
@@ -0,0 +1,9 @@
+@@ -1,3 +1,4 @@
++setupRequired(scons)
+ envPrepend(PATH,${PRODUCT_DIR}/bin)
+ envPrepend(LD_LIBRARY_PATH,${PRODUCT_DIR}/lib)
```
Should have DYLD_LIBRARY_PATH as well here.

ktlim commented on patches/tmvTable.patch:
```diff
@@ -0,0 +1,9 @@
++setupRequired(scons)
+ envPrepend(PATH,${PRODUCT_DIR}/bin)
+ envPrepend(LD_LIBRARY_PATH,${PRODUCT_DIR}/lib)
+ envPrepend(LIBRARY_PATH,${PRODUCT_DIR}/lib)
```
What uses this variable?  It's not usually present.

ktlim commented on ups/eupspkg.cfg.sh:
```diff
@@ -0,0 +1,2 @@
+NJOBS=1
+export SCONSFLAGS=$SCONSFLAGS" USE_UNKNOWN_VARS=true PREFIX="$PREFIX"/lib"
```
It's unusual to have a PREFIX point to a lib directory.  You should at least have a comment to explain this.

ktlim commented on ups/tmv.table:
```diff
@@ -0,0 +1,7 @@
+setupRequired(scons)
+envPrepend(PATH,${PRODUCT_DIR}/bin)
+envPrepend(LD_LIBRARY_PATH,${PRODUCT_DIR}/lib)
```
Add DYLD_LIBRARY_PATH here.

ktlim commented on ups/tmv.table:
```diff
@@ -0,0 +1,7 @@
+setupRequired(scons)
+envPrepend(PATH,${PRODUCT_DIR}/bin)
+envPrepend(LD_LIBRARY_PATH,${PRODUCT_DIR}/lib)
+envPrepend(LIBRARY_PATH,${PRODUCT_DIR}/lib)
```
LIBRARY_PATH, C_INCLUDE_PATH, and CPATH are not typically used.  Are you sure they are necessary?

danielsf committed:
fixed tmvTable.patch to remove unneeded environment variables and
add DYLD_LIBRARy_PATH

danielsf committed:
fixed tmv.table file the same way I fixed the tmv table patch

danielsf commented:
Okay.  I wasn't sure what our policy was on changing the source code of external packages.

I agree that just removing the externally-provided table files will make life easier.

danielsf committed:
removed lib from PREFIX in eupspkg.cfg.sh

ktlim commented:
Does TMV come with the ups directory as part of its distribution?  If so, then forget what I said (but it is unusual).

ktlim commented:
Oh, I see, it's from Mike Jarvis.  That makes more sense.  I guess you may have to go back to what you were doing, if the packaging ups directory doesn't override the upstream ups directory.  Sorry.

danielsf commented:
I am currently trying a build in which I deleted the native table file in the tarball.

The build seems to be working.  Assuming it works, is there any harm in providing a tarball from which we have extracted the native table file (I guess, if we ever upgrade to a future version, whoever does that will have to be warned that they will have to get rid of the native table file)?

ktlim commented:
Yes, the problem is future upgrades.  We should leave the upstream distribution as is, to the extent possible.  You might recommend that your patches be applied by TMV so that we can remove them from our packaging.

danielsf commented:
Okay

danielsf committed:
set environment variable TMV_DIR for GalSim's use

danielsf commented:
On a similar note:

The LIBRARY_PATH, C_INCLUDE_PATH, and CPATH variables are all set by the native table file (which is why I included them in our table file).

I am testing a build in which I do not include them in our table file and I use the patches to remove them from the native table file.

If that passes, should I go with that, or err on the side of setting the environment variables?

ktlim commented:
I would say that we should not unnecessarily set variables (although setting LD_LIBRARY_PATH on Darwin and DYLD_LIBRARY_PATH on non-Darwin violates that), so keep the patch.

danielsf committed:
add TMV_DIR to the table file patch

danielsf committed:
remove unnecessary (and confusing) prepend statement
